hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/access/AbstractAccessDelegate.java

```diff
@@ -108,11 +108,12 @@ public boolean putFromLoad(final Object key, final Object value, final long txTi
         return putFromLoad(key, value, txTimestamp, version);
     }
 
-    /**
-     * This is an asynchronous cache access strategy.
-     * NO-OP
-     */
     public void remove(final Object key) throws CacheException {
+        try {
+            cache.remove(key);
+        } catch (HazelcastException e) {
+            throw new CacheException("Operation timeout during remove operation from cache!", e);
+        }
```

Comment:
OK, I needed to rewrite more than I thought.  To be totally consistent, only "synchronous" regions can modify the cache immediately on `insert`, `update`, and `remove`.  "Asynchronous" regions must "softlock" the cache entry, treat `insert`, `update`, and `remove` as a no-op, and apply the change on `afterInsert`, `afterUpdate` or `unlockItem`, to maintain proper READ_COMMITTED isolation. 

For `EntityRegionAccessStrategy`, the contract is:
```
 * Contract for managing transactional and concurrent access to cached entity
 * data.  The expected call sequences related to various operations are:<ul>
 *     <li><b>INSERTS</b> : {@link #insert} -> {@link #afterInsert}</li>
 *     <li><b>UPDATES</b> : {@link #lockItem} -> {@link #update} -> {@link #afterUpdate}</li>
 *     <li><b>DELETES</b> : {@link #lockItem} -> {@link #remove} -> {@link #unlockItem}</li>
 *     <li><b>LOADS</b> : {@link @putFromLoad}</li>
 * </ul>
 * <p/>
 * There is another usage pattern that is used to invalidate entries
 * after performing "bulk" HQL/SQL operations:
 * {@link #lockRegion} -> {@link #removeAll} -> {@link #unlockRegion}
```

For `CollectionRegionAccessStrategy`, the contract is:

```
 * Contract for managing transactional and concurrent access to cached collection
 * data.  For cached collection data, all modification actions actually just
 * invalidate the entry(s).  The call sequence here is:
 * {@link #lockItem} -> {@link #remove} -> {@link #unlockItem}
 * <p/>
 * There is another usage pattern that is used to invalidate entries
 * after performing "bulk" HQL/SQL operations:
 * {@link #lockRegion} -> {@link #removeAll} -> {@link #unlockRegion}
```

This is similar enough that I didn't see much benefit in splitting out the `Collection` and `Entity` `AccessDelegate`s into separate classes. 

The real problem was that the `LocalRegionCache` implementation of (the unfortunately named) `unlockItem` and `unlockRegion` did no invalidations on other nodes, so Collections in particular were prone to going stale on other cluster nodes.  I've done this plus straightened out the various `Delegate` classes to be more consistent and in line with ehcache.